pkg/controllers/managementuser/helm/controller.go

```diff
@@ -336,6 +338,32 @@ func (l *Lifecycle) Run(obj *v3.App, template string, tempDirs *hCommon.HelmPath
 	return l.createAppRevision(obj, template, notes, false)
 }
 
+// pruneOldRevisions checks if the total number of app revisions will not exceed the set maximum.
+// If it will, the method deletes the oldest revision(s).
+func (l *Lifecycle) pruneOldRevisions(obj *v3.App) {
+	if obj.Spec.MaxRevisionCount < 1 {
+		obj.Spec.MaxRevisionCount = defaultMaxRevisionCount
+	}
+	_, projectName := ref.Parse(obj.Spec.ProjectName)
+	appRevisionClient := l.AppRevisionGetter.AppRevisions(projectName)
+	revisions, err := appRevisionClient.List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", appLabel, obj.Name),
+	})
+	if err != nil {
+		logrus.Tracef("[helm-controller] Failed to list app revisions for app %s: %v", projectName, err)
+	}
+
+	if len(revisions.Items) >= obj.Spec.MaxRevisionCount {
+		logrus.Tracef("App %s is exceeding the maximum (%d) number of stored revisions", obj.Name, obj.Spec.MaxRevisionCount)
+		deleteCount := len(revisions.Items) - obj.Spec.MaxRevisionCount
+		for _, revision := range revisions.Items[:deleteCount] {
```

Comment:
Are we ok with not iterating over the revision lists when `len(revisions.Items) == max`? I see the "exceeding the maximum" log so I am curious.
https://play.golang.org/p/j-ChGHx-hco

Reply:
Nick, you won't believe it, but I wrote the exact same Playground example before opening the PR to test this Go slice behavior! Wow!
But yeah, now that the pruning is done after a new revision is created, I should update it to just be `len(revisions.Items) > max` and not `>=`.

Reply:
This behavior is also interesting:
https://play.golang.org/p/eOh1Yh8KqjN